veps/sig-network/link-state-proposal.md

```diff
@@ -0,0 +1,232 @@
+In case the field's value is `InterfaceStateLinkDown`, the interface will be created as follows:
+```xml
+<interface>
+  ...
+  <link state='down'/>
+  ...
+</interface>
+```
+
+> [!NOTE]
+> For additional details please see libvirt's [documentation](https://libvirt.org/formatdomain.html#modifying-virtual-link-state).
+
+virt-launcher's `LibvirtDomainManager.SyncVMI` will be extended to support updating the interfaces link state while the VM is running.
+
+The new logic will:
+1. Fetch the domain.
+2. For each interface in the VMI spec:
+ - Compare the actual link state against the desired link state.
+ - Adjust the interface entry in the domain in case there is a need to take action.
+
+virt-launcher's NIC hotplug logic will be adjusted to support hot-plugging a NIC while its link state is down.
+
+### Network Binding Plugins Support
+
+When using the functionality purposed by this enhancement with network binding plugins using a sidecar container -
+it is up to the plugin's vendor to support this functionality - as they control the interface section in the domain XML.
+
+### Link State Reporting
+
+A new field will be added to the VMI interface status that will be used for reporting the interface's current link state (up/down):
+```go
+type VirtualMachineInstanceNetworkInterface struct {
+	...
+	LinkState string `json:"linkState,omitempty"`
+	...
+}
+```
+
+virt-handler will be responsible for reporting this field's value, based on domain spec information received from virt-launcher.
+An empty specification (as exists today) would be considered as `up`.
+
+SR-IOV NICs' link state will not be reported, as it could change without KubeVirt awareness.
+
+> [!NOTE]
+> The linkState information will be reported for interfaces which are present in the VirtualMachineInstance's spec.
+> Interfaces defined inside the guest will not have their link state reported.
+>
+> Changing the link state of an interface inside the guest or in the underlying network provider - will not affect this report:
+> 
+> 1. KubeVirt is not aware of network provider internal state.
+> 2. qemu guest agent does not report the interface link states inside the guest. 
+
+
+## API Examples
+### Controlling The Link State
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: my-vm
+spec:
+  template:
+    spec:
+      domain:
+        devices:
+          interfaces:
+            - name: default
+              state: down
+              masquerade: { }
+      networks:
+        - name: default
+          pod: { }
+```
+### Link State Reporting
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  name: my-vm
+spec:
+  domain:
+    devices:
+      interfaces:
+      - name: default
+        state: down
+        masquerade: { }
+  networks:
+  - name: default
+    pod: { }
+status:
+  interfaces:
+    - name: default
+      linkState: down
+```
+
+## Alternatives
```

Comment:
Nit: I think it's better to remove empty sections. (or at least add a line explaining why it's irrelevant like below)

Reply:
It was removed.